packages/strapi-design-system/src/NumberInput/NumberInput.js

```diff
@@ -51,11 +51,22 @@ export const NumberInput = React.forwardRef(
     const generatedId = useId('numberinput', id);
     const numberParserRef = useRef(new NumberParser(getDefaultLocale()));
     const numberFormaterRef = useRef(new NumberFormatter(getDefaultLocale()));
+    const inputRef = useRef();
 
     if (!label && !props['aria-label']) {
       throw new Error('The NumberInput component needs a "label" or an "aria-label" props');
     }
 
+    useEffect(() => {
+      if (inputRef.current) {
+        setInputValue(value === undefined || value === null ? INITIAL_VALUE : String(value));
```

Comment:
I think we would need more value verification before updating `inputValue` 🤔
for example I used a button to update the value to a string: 

<img width="572" alt="image" src="https://user-images.githubusercontent.com/71838159/179536325-46cdb7fb-cdda-4453-81db-c558653ffbb3.png">

Reply:
Actually its not that easy for me as for back-end dev (with angular experience). Maybe you can give me an advice. 

For validation we got `numberParserRef` but we can't use it in `useEffect(..., [])` and on initial call ref is still not initialized but we can't just check if its initialized or not (with `if (numberParserRef.current)`) and should wait for init so starting value will be checked and converted correctly. 

So i can spend time on weekend to resolve this or do not fix it now, open new issue and let core team fix it later on.

Reply:
hi again 🌞 sorry for the delay!
I worked on finding a fix this morning - we have access to `numberParserRef` in the `useEffect` but since `value` is received as a number instead of a string that's why it seems like not working, the value needs `value.toString()` transformation before being passed to `numberParserRef.current.isValidPartialNumber()` function.
I think it would be great to have a function able to handle the change both for `handleChange` and the `useEffect` - I could try to push on your branch or let you try to handle it as you prefer 😊